dbaccess/dbtx.go

```
@@ -0,0 +1,75 @@
+package dbaccess
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/kaspanet/kasparov/database"
+	"github.com/kaspanet/kasparov/httpserverutils"
+)
+
+// Context is an interface type representing the context in which queries run, currently relating to the
+// existance or non-existance of a database transaction
+// Call `.NoTx()` or `.NewTx()` to aquire a Context
+type Context interface {
+	db() (*gorm.DB, error)
```

instead of returning an error you can preload the databse NewTx/NoTx, and return error in there